### ISSUE TITLE
[BUG] GIL is not released in some KMS sync wrappers

### ISSUE THREAD

**Issue by leandrodamascena** [ferrumio/pydynox#281]
### Description

`sync_encrypt_impl` and `sync_decrypt_impl` in `pydynox/src/kms/operations.rs` call `runtime.block_on()` without releasing the GIL.

### Steps to Reproduce

```python
import time
from concurrent.futures import ThreadPoolExecutor
from pydynox import DynamoDBClient

client = DynamoDBClient()
kms = client.kms_encryptor(key_id="alias/my-key")

def encrypt_value(i):
    start = time.time()
    result = kms.sync_encrypt(f"secret-{i}")
    return time.time() - start

# Run 10 encryptions in parallel using threads
with ThreadPoolExecutor(max_workers=10) as pool:
    results = list(pool.map(encrypt_value, range(10)))

# All 10 run one at a time because the GIL is held during block_on()
```

### Expected Behavior

When using `ThreadPoolExecutor`, KMS sync operations should run in parallel. The GIL should be released during the KMS API call.

### Actual Behavior

All requests run one at a time. The GIL is held during the entire `runtime.block_on()` call:

```rust
// kms/operations.rs — no py parameter, can't release GIL
pub fn sync_encrypt_impl(
    client: &Client,       // no py: Python<'_> parameter!
    runtime: &Arc<Runtime>,
    key_id: &str,
    context: &HashMap<String, String>,
    plaintext: &str,
) -> PyResult<(String, KmsMetrics)> {
    runtime.block_on(execute_encrypt(...))  // GIL held!
}
```

### Environment

- pydynox version: 0.26.0
- Python version: 3.11+
- OS: any
- AWS Region: any

### Additional Context

Same fix as bug #280: add `py: Python<'_>` and wrap with `py.detach()`.
